system/include/emscripten/bind.h

```diff
@@ -1938,6 +1962,19 @@ namespace internal {
 
 template<typename MapType>
 struct MapAccess {
+#if __cplusplus >= 201703L
+    static std::optional<typename MapType::mapped_type> get(
+        const MapType& m,
+        const typename MapType::key_type& k
+    ) {
+        auto i = m.find(k);
+        if (i == m.end()) {
+            return {};
+        } else {
+            return i->second;
+        }
+    }
+#else
```

Comment:
Wouldn't this be a breaking change? I'm not sure what the policy for those are in Embind, but seems somewhat risky as users who already accessed values via the old signature, will have their code not compile after version upgrade. That switching C++ versions changes the signature seems even more risky.

Maybe safer to introduce the new method as `get_optional` / `maybe_get` / something like that, and keep the old one unconditionally there?

Reply:
User's wouldn't be directly calling this code. Only the JS code will be invoking this and the return values will be exactly the same as before in JS.

Reply:
Ah... but if it's only called internally by JS, I'm not sure I understand the value of duplicated implementation code? If it cleaned up `val` in favour of `optional`, sure, but if we can't remove it, then keeping the `val` path seems a bit easier?

Reply:
If we don't use `optional` here then we don't get improved type of `number | undefined` for typescript when using c++17. I'm hoping at some point we can drop c++11 support and the val code will go away.

Reply:
Oh I see, that makes sense.

Can you add an explanatory comment to those modified places (something along the lines "this branch is used for generating more specific TypeScript types" or something) as I imagine someone looking at the code in the future will be also confused about seemingly duplicate functionality.

I'll approve PR now regardless, I just think a comment will be useful to avoid future confusion.